bokehjs/src/coffee/models/renderers/glyph_renderer.coffee

```diff
@@ -230,7 +230,10 @@ export class GlyphRendererView extends RendererView
     @glyph.draw_legend_for_index(ctx, x0, x1, y0, y1, index)
 
   hit_test: (geometry) ->
-    @glyph.hit_test(geometry)
+    if @model.visible
+      return @glyph.hit_test(geometry)
+    else
+      return null
```

Comment:
Right I missed `@model` / `@glyph` dichotomy. My point is this needs a test, to prevent regressions. This would be both fairly easy to break and fairly hard to notice given our existing tests.